cognite/neat/core/workflow/manager.py

```diff
@@ -3,15 +3,30 @@
 import logging
 import os
 import sys
+import time
 import traceback
 from pathlib import Path
 
 from cognite.client import CogniteClient
+from prometheus_client import Gauge
+from pydantic import BaseModel
 
 from cognite.neat.core.workflow import BaseWorkflow
 from cognite.neat.core.workflow.base import WorkflowDefinition
+from cognite.neat.core.workflow.model import FlowMessage, InstanceStartMethod, WorkflowState
 from cognite.neat.core.workflow.tasks import WorkflowTaskBuilder
 
+live_workflow_intances = Gauge("neat_workflow_live_instances", "Count of live workflow instances", ["itype"])
+
+
+class WorkflowStartStatus(BaseModel):
+    workflow_instance: BaseWorkflow = None
```

Comment:
This seems like a strange design. Why do a workflow status need access to the workflow instance?

Reply:
As I see it used in the code, if you return the wokflow instance from the `start_workflow `method. You will have it available in to do at least where the `workflow_instance `attribute in the `WorflowStartStatus `is used in the code today.  

Here is the place in the code, base.py line 260.

```python
                    workflow_task = self.task_builder.start_workflow_task(
                        workflow_name=step.params.get("workflow_name", ""), sync=sync, flow_message=self.flow_message
                    )
                    if workflow_task.status.is_success and workflow_task.state == WorkflowState.COMPLETED:
                        new_flow_message = workflow_task.status..flow_message
                    else:
                        logging.error(f"Workflow step {step.id} failed to start workflow task")
                        if workflow_task.status.is_success:
                            raise WorkflowStartException(workflow_task.last_error)
                        else:
                            raise WorkflowStartException(workflow_task.status.status_text)
```

instead of

```python
                    start_status = self.task_builder.start_workflow_task(
                        workflow_name=step.params.get("workflow_name", ""), sync=sync, flow_message=self.flow_message
                    )
                    if start_status.is_success and start_status.workflow_instance.state == WorkflowState.COMPLETED:
                        new_flow_message = start_status.workflow_instance.flow_message
                    else:
                        logging.error(f"Workflow step {step.id} failed to start workflow task")
                        if start_status.is_success:
                            raise WorkflowStartException(start_status.workflow_instance.last_error)
                        else:
                            raise WorkflowStartException(start_status.status_text)
```

Reply:
Anyways, there might be more context I am missing, so I will approve and it is up to you whether to do more changes based on the comment above.